xbmc/filesystem/FileDirectoryFactory.cpp

```diff
@@ -64,20 +64,22 @@ IFileDirectory* CFileDirectoryFactory::Create(const CURL& url, CFileItem* pItem,
 
   std::string strExtension=URIUtils::GetExtension(url);
   StringUtils::ToLower(strExtension);
-  VECADDONS codecs;
-  CAddonMgr::GetInstance().GetAddons(codecs, ADDON_AUDIODECODER);
-  for (size_t i=0;i<codecs.size();++i)
+  if (!strExtension.empty())
```

Comment:
the issue is android only, even the title of this PR stats this. again, this workaround will break with the mentioned PR when addons can extend vfs